newIDE/app/src/Utils/UseOpenInitialDialog.js

```diff
@@ -37,18 +39,36 @@ const useOpenInitialDialog = ({
           });
           removeRouteArguments(['initial-dialog']);
           break;
+        case 'login':
+          const connectionId = routeArguments['connection-id'];
+          const loginEnvironment =
+            routeArguments['login-environment'] === 'dev' ? 'dev' : 'live';
+          // Add timeout to give time to the app to sign in with Firebase
+          // to make sure the most relevant dialog is opened.
```

Comment:
Any chance we could somehow wait for a auth state change?

Reply:
That's a good question. I tried without the timeout and it didn't work. And I'm not sure how to get a signal when firebase is done initializing.

Reply:
Timeout will work for now!